server/handlers/handlers.go

```diff
@@ -388,6 +390,24 @@ func (ah *ActivityHandler) handleCreatedActivity(msg *clientmodels.Message, subs
 		}
 	}
 
+	// Update the last chat received at for the participants of DM/GM
+	if chat != nil && len(userIDs) > 0 {
+		// exclude the sender from the list of participants
+		participants := []string{}
+		for _, userID := range userIDs {
+			if userID != post.UserId {
+				participants = append(participants, userID)
+			}
+		}
+
+		if len(participants) > 0 {
+			err := ah.plugin.GetStore().SetUsersLastChatReceivedAt(participants, post.CreateAt*1000)
```

Comment:
times are `*1000` because Mattermost uses milliseconds, but we use microseconds in the plugin.

Reply:
Is there a way to make the `*1000` explicit? (i.e. using some `time` method?)

Reply:
Using go's `time` package would be quite ugly:

```go
(time.Duration(post.CreateAt) * time.Millisecond).Microseconds()
```

But for the sake of making this more clear, I'm ok with creating a helper `MilliToMicroSeconds()` that does the `*1000`

Reply:
All good, only 0/5 here -- whatever is clearest.